scripts/screeps_rl_experiment_card.py

```
@@ -145,15 +239,259 @@ def build_card(
+        variant = registry[variant_id]
+        parameters = construction_priority_parameters_or_none(variant.parameters)
+        if parameters is None:
+            raise CardValidationError(
+                f"strategy registry {repo_relative(registry_path)} variant {variant_id} "
+                "must define finite construction-priority parameters"
+            )
+        defaults[variant_id] = parameters
+    return defaults
+
+
+def construction_priority_parameters_or_none(raw: Any) -> JsonObject | None:
```

Ruff (0.15.12)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `raw`

(ANN401)